adapter/slack.py

```diff
@@ -135,21 +150,31 @@ async def list_bots(self, request: Request):
         except SlackApiError as e:
             raise HTTPException(status_code=400, detail=f"Slack API Error : {e}")
     
-    def event_message(self, event, say):
+    def event_message(self, event):
         if 'thread_ts' in event:
-            self.event_message_replied(event, say)
+            self.event_message_replied(event)
     
-    def event_message_replied(self, event, say):
+    def event_message_replied(self, event):
         parent_message = self.app.client.conversations_history(
             channel=event['channel'],
             inclusive=True,
             oldest=event['thread_ts'],
-            limit=1
+            limit=1,
+            include_all_metadata=True
         )
         parent_user, parent_text = parent_message['messages'][0]['user'], parent_message['messages'][0]['text']
         
         if parent_user == self.app_user_id and 'asked' in parent_text:
-            self.bot_replied(event, say)
+            self.bot_replied(event, parent_message)

```

Comment:
Hey, now that there is a metadata for each `/ask` message, don't you think this check is better refactored by checking if the relevant metadata exists rather than checking if there is an 'asked' text in the message?

Reply:
I agree, that's a nice catch! 👌